server/controllers/updateEvent.js

```diff
@@ -1,13 +1,17 @@
 const { Event } = require('../database/models');
 
 exports.updateEvent = (req, res) => {
-  Event.update({ status: 1 }, { where: { id: req.params.id } })
-    .then((result) => {
-      if (result) {
-        res.json({ success: 'status updated' });
-      } else {
-        res.json({ sucess: 'status did not get updated!' });
-      }
-    })
-    .catch(() => res.status(500).json({ error: 'something wrong in query ' }));
+  if (req.userRole === 'admin') {
```

Comment:
I would think about have a middleware for permissioning rather than doing it on this endpoint, so the middle ware would just be something like:

```js
allowAdmin = (req, res, next) => {
  if (req.userRole === 'admin') {
    next()
  } else {
       res.status(401).json({ error: 'not auth' });
  }
}

```

and then this handler would just do the update